qiskit_experiments/framework/base_experiment.py

```diff
@@ -362,6 +360,11 @@ def circuits(self) -> List[QuantumCircuit]:
         # values for any explicit experiment options that affect circuit
         # generation
 
+    def transpiled_circuits(self):
```

Comment:
This is how originally I implemented this in https://github.com/Qiskit/qiskit-experiments/pull/404#discussion_r722435547 but @chriseclectic recommended to me to integrate everything in a single public method rather than having two for logical and physical circuits.

Reply:
Please add docstring here. I understand that this is provided as a public function to give child classes a way to directly generate physical circuits (transpiled circuits) instead of logical circuits (pre-transpiled). I think such an intension should be clearly documented here. In fact, after this PR, Experiment class developers may not need to implement `circuits()` anymore as long as they implement `transpiled_circuits()`. I like this function but this is a big change to be reviewed carefully.

Reply:
+ how about caching the transpiled circuits? I often visualize transpiled circuits before running an experiment. In that case, `transpiled_circuits()` is called twice. With caching, we can speed up the second run.